### DISCUSSION TITLE
Idea: A natural language documentation chatbot

### DISCUSSION THREAD

**Discussion by Ilikepizza2** [vercel/next.js#54832] [Ideas]
### Goals

1. To add a LLM based chatbot which can answer about all the new things from the documentation in natural language.
2. To make this as an addition to the already present quick search.
3. As an example, you can look at [langchain docs](https://js.langchain.com/docs/)

### Non-Goals

1. This is not a replacement of quick search
2. This doesn't aim to replace any other feature.

### Background

Idea: Since we are already updating the documentation, why not we add a Natural Language Processing based Chatbot to the documentation which can query the docs itself and answer in natural language. This feature would not replace the `ctrl+K` feature but would instead be an addition to the already present static search. One prime example would be langchain docs: https://js.langchain.com/docs/ Just press `ctrl+k` and the mendable ai would popup.

I have worked on very similar things before and am open to discussions as well as possible solutions on how to implement this. If this feature is approved, then it would help since a lot of us developers have a hard time searching and relating the two documentations.

Sorry if this isn't the correct format to ask questions. I'm quite new here.

### Proposal

This would be easily possible with any vector database, any large language model and an embeddings model. I've worked on very similar things before and if this feature is approved, I'm confident that I can handle this feature.

**Comment by AcidicArmadillo** [comment]
Good idea

**Comment by RhyshaKachari** [comment]
it would indeed be very useful 🤔

**Comment by axshatj** [comment]
This proposal to add a Natural Language Processing-based chatbot to documentation is a great idea, enhancing user experience without replacing existing features. Adding user analytics to track how often the chatbot is used and which questions it struggles with could provide valuable insights for continuous improvement.

**Reply by Ilikepizza2** [reply]
That's also a great idea. Never thought about data analysis on documentation!

**Comment by notabhishek22** [comment]
Idea seems quite attractive. Would be very helpful if properly integrated 👍

**Comment by nikolailehbrink** [comment]
Could be even quite easily achieved by using for example [inkeep](https://inkeep.com/) which is used in the [bun docs](https://bun.sh/docs). I think that this would make a significant difference for developers and it's just a logical step for better documentation.

**Comment by sreeharsha-rav** [comment]
This is a good idea, to support this I came across Orama which provides an LLM chat and it's been implemented in Deno docs as well as SolidJS docs. I have opened up a discussion here [Integrate AI-Powered Search Using Orama in Next.js Documentation #68905](https://github.com/vercel/next.js/discussions/68905).
